File: app/src/main/java/com/verbatoria/di/late_send/LateSendComponent.java
package com.verbatoria.di.late_send;

import com.verbatoria.di.session.SessionModule;
import com.verbatoria.di.session.SessionScope;
import com.verbatoria.presentation.late_send.view.LateSendActivity;

import dagger.Subcomponent;

/**
 * Компонент Даггера для логина
 *
 * @author nikitaremnev
 */
@Subcomponent(modules = {LateSendModule.class, SessionModule.class})
@SessionScope
public interface LateSendComponent {
    
    void inject(LateSendActivity lateSendActivity);

}

File: app/src/main/java/com/verbatoria/di/late_send/LateSendComponent.kt
package com.verbatoria.di.late_send;

import com.verbatoria.di.BaseInjector
import com.verbatoria.ui.late_send.LateSendActivity
import dagger.Subcomponent

/**
 * @author n.remnev
 */

@Subcomponent(modules = [LateSendModule::class])
interface LateSendComponent : BaseInjector<LateSendActivity> {

    @Subcomponent.Builder
    interface Builder {

        fun build(): LateSendComponent

    }

}
